### PR TITLE
Moves device analyzer's scanning to preattack

### LINKED ISSUE #3052: Mechanic's Device Analyzer can no longer scan Autolathes
Previously the device analyzer and Mechanic's PDA device analyzer feature could scan Autolathes for use in producing blueprints through the reverse engine. No longer possible.

### PR DESCRIPTION
**Pull request by 9600bauds**
Fixes #3052
This seems to work a LOT better than having it in afterattack. You'll always be able to scan whatever you want to scan, the autolathe won't try to eat your scanner, no need for snowflake code so you don't place it on top of that conveyor belt, you won't put it in that bag of holding while trying to scan it.
And if you actually want to put the scanner in that bag of holding, it will still go in if it doesn't successfully scan it, so just try to scan it twice and it will go.

### CODE REVIEW

**Comment by clusterfack** [comment]
I dont think was preattack because I dont think preattack even existed back when this was first made. I believe this is a pretty good application of it though.

File: code/modules/research/mechanic/device_analyser.dm [hunk 1]
```diff
@@ -26,7 +26,7 @@
 	loadone = !loadone
 	usr <<"<span class='notice'> You set the Device Analyzer to [loadone ? "transfer one design" : "transfer all designs"] on use.</span>"
 
-/obj/item/device/device_analyser/afterattack(var/atom/A, mob/user, proximity_flag) //Hurrah for after-attack
+/obj/item/device/device_analyser/preattack(var/atom/A, mob/user, proximity_flag) //Hurrah for after-attack
 	if(get_turf(src) != get_turf(user)) //we aren't in the same place as our holder, so we have been moved and can ignore scanning
```
Comment (clusterfack) [inline code/modules/research/mechanic/device_analyser.dm]: Is... this necessary using the proximity flag?

Comment (ComicIronic) [inline code/modules/research/mechanic/device_analyser.dm]: No. It's not a possibility, and was included originally to stop the message showing up whenever the user put the item on a table.

**Comment by ComicIronic** [comment]
Preattack implementations are rather dangerous because preattack overrides all other clickcode if it succeeds, which is annoying for handling certain scenarios.

**Comment by ComicIronic** [comment]
I'm worried that you didn't test this with respect to the Reverse Engine. You have to have a scan of the Reverse Engine in the analyser for scanning it to fail and for the attackby code to call.

**Comment by 9600bauds** [comment]
Oh whoops, you're right. Will fix.

**Comment by 9600bauds** [comment]
Actually the PDA's device analyzer function is still an afterattack which results in not being able to scan the reverse engine with it, not sure how could this be handled. Copy the functionality as a new preattack for the PDA?

**Comment by ComicIronic** [comment]
Are you returning 1 on the PDA preattack when the DA preattack is returning 1?

**Comment by 9600bauds** [comment]
Short and to the point, works perfectly now. Thank you very much Comic.

**Pull request merged**